src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1400,7 +1398,7 @@ private new void VisitCollectionElementInitializer(BoundCollectionElementInitial
 
         private void SetResult(BoundExpression node)
         {
-            _resultType = TypeSymbolWithAnnotations.Create(node.Type);
+            ResultType = TypeSymbolWithAnnotations.Create(node.Type);
```

Comment:
This is the case I'm thinking about when I asked about knowing if it's a valid lvalue before. There is no understanding here whether this is or isn't a valid lvalue. Yet we're presenting the type now as a `LValueType`.

Reply:
I'll take another look at an alternative and will get back to you.

---
In reply to: [258142088](https://github.com/dotnet/roslyn/pull/33447#discussion_r258142088) [](ancestors = 258142088)

Reply:
Changed the `VisitResult` to hold a `default` `LValueType` for expressions that aren't l-values.

---
In reply to: [258232704](https://github.com/dotnet/roslyn/pull/33447#discussion_r258232704) [](ancestors = 258232704,258142088)